newIDE/app/src/EventsSheet/index.js

```diff
@@ -1084,406 +1115,427 @@ export default class EventsSheet extends React.Component<Props, State> {
-                        />
-                        {this._renderInstructionEditorDialog(
-                          // Force using the new instruction editor on touch screens.
-                          preferences.useNewInstructionEditorDialog ||
-                            screenType === 'touch'
-                        )}
-                        {this.state.analyzedEventsContextResult && (
-                          <EventsContextAnalyzerDialog
-                            onClose={this._closeEventsContextAnalyzer}
-                            eventsContextResult={
-                              this.state.analyzedEventsContextResult
-                            }
-                          />
+                    <EventsTree
+                      ref={eventsTree => (this._eventsTree = eventsTree)}
+                      key={events.ptr}
+                      onScroll={this._ensureFocused}
+                      events={events}
+                      project={project}
+                      scope={scope}
+                      globalObjectsContainer={globalObjectsContainer}
+                      objectsContainer={objectsContainer}
+                      selection={this.state.selection}
+                      onInstructionClick={this.selectInstruction}
+                      onInstructionDoubleClick={this.openInstructionEditor}
+                      onInstructionContextMenu={this.openInstructionContextMenu}
+                      onAddInstructionContextMenu={
+                        this.openAddInstructionContextMenu
+                      }
+                      onAddNewInstruction={this.openInstructionEditor}
+                      onPasteInstructions={
+                        this.pasteInstructionsInInstructionsList
+                      }
+                      onMoveToInstruction={this.moveSelectionToInstruction}
+                      onMoveToInstructionsList={
+                        this.moveSelectionToInstructionsList
+                      }
+                      onParameterClick={this.openParameterEditor}
+                      onEventClick={this.selectEvent}
+                      onEventContextMenu={this.openEventContextMenu}
+                      onAddNewEvent={(
+                        eventType: string,
+                        eventsList: gdEventsList
+                      ) => {
+                        this.addNewEvent(eventType, {
+                          eventsList,
+                          indexInList: eventsList.getEventsCount(),
+                        });
+                      }}
+                      onOpenExternalEvents={onOpenExternalEvents}
+                      onOpenLayout={onOpenLayout}
+                      searchResults={eventsSearchResultEvents}
+                      searchFocusOffset={searchFocusOffset}
+                      onEventMoved={this._onEventMoved}
+                      showObjectThumbnails={
+                        this.props.userPreferences.values
```

Comment:
uber nitpicking: you could `..., userPreferences } = this.props;` at the beginning of the render to avoid this.props.userPreferences everywhere and reduce a bit the diff here.